### DISCUSSION TITLE
Advice of settings AIO fans rpm depending on CPU temp as main and GPU as second

### DISCUSSION THREAD

**Discussion by godzinilla** [Rem0o/FanControl.Releases#2453] [Q&A] [answered]
Hello,

I am coming from the full air cooled PC to the AIO (for the CPU).

My previous setup was:
- 3 intake fans on the front
- 2 exhaust fans on the top
- 1 exhaust fan on the back
- single fan CPU tower cooler

Currently there is:
- 3 intake fans on the front
- 1 exhaust fan on the back
- AIO with 2 exhaust fans on the top

Peripherals:
- GPU: RX 6950XT
- CPU: 12400F

In the previous setup I set the intake slightly faster than the exhaust to set positive chamber. The curves are set as automatic with following settings:

- GPU Average as a temp source for INTAKE and EXHAUST - average of GPU hotspot and GPU core
- INTAKE: min. temp 51, max. temp 90, min. speed 20, max. speed 100, hysteresis 2 deg., response time 2 sec.
- EXHAUST: min. temp 51, max. temp 90, min. speed 18, max. speed 100, hysteresis 2 deg., response time 2 sec.
- CPU: Core Average temp; min. temp 32, max. temp 60, min. speed 15, max. speed 70, hysteresis 2 degrees, response time 2 sec.

So the EXHAUST fans depends mainly on the GPU temperature.

![fans](https://github.com/Rem0o/FanControl.Releases/assets/77450480/2173630b-e984-4386-86e7-e4079b0c2af9)

Right now the AIO fans are set only for the CPU temps and not related to the GPU temperature, so I have only one EXHAUST fan depending on the GPU temperature.

I need an advice how to set the AIO fans to double act as a EXHAUST fan for the CPU and GPU temperature. My PC is more in idle state than full throttle so I think it should be set to switch to the GPU temp from some point. What option in FanControl should I use to get the easiest and best result?

**Comment by ionGL** [comment]
You should use a MIX fan curve, with the max function.

**Reply by godzinilla** [reply]
But with the MAX function the AIO will always use the GPU temps because CPU in idle is about 30 degrees and GPU 50-55 (with 0 rpm function below 55 degrees). Would be better to switch to the GPU temps from the defined temp eg. 60 degrees.

**Reply by ionGL** [reply]
Here is how you do it:

- Have a new graph curve (temperature source gpu average) with a 0% fan speed at 55 degrees and whatever % you think you want at 60C, 65 etc.
- Another graph curve (temperature source core average), probably the one called CPU in your screenshot.

Now you add a MIX curve and combine in it both curves mentioned. You then select the MIX curve in the control for the aio fans. When the gpu is bellow 55C, the mix curve will set the cpu curve %. Once the gpu goes up, it will switch to the other curve. It's a beautiful thing!

**Reply by Rem0o** [reply]
^ What he said.

**Reply by godzinilla** [reply]
Ok but this will work only when the GPU will be running in the 0 rpm fan mode with temps below 55 degrees - the 55 value is hardcoded and cannot be changed. My GPU regular standby temps fluctuating between 50-56 degrees so the GPU fans regularly ramps in to cool the card and the minimum fans speed is 20%.

The 15% for the CPU is good enough to keep it average 30 degrees.

If I disable the 0 rpm mode and use proposed MIX curve the AIO fans will always operate based on the GPU temps (in standby of course).

So it will work but only in this certain situation.

And it all of that I want to create positive pressure in my case.

My current settings:

![fans2](https://github.com/Rem0o/FanControl.Releases/assets/77450480/62eec467-982f-4aaa-8315-445f13d42a78)

If managed to get the EXHAUST and AIO fans on the same level (I believe the real performance with just pulling the hot air out will differ between single EXHAUST fan and two AIO fans so finally the AIO should be slightly faster).

The INTAKE is slightly faster.

The GPU temps looks interesting. Not sure it's because the system is running only for about the hour or the cooling situation in my case is better without the tower cooler.

**Reply by ionGL** [reply]
You didn't pay attention to my reply. If your aio fans are controlled by the "Cpu Fan" control, you must select the mix curve there.

Give up on all linear and auto curves, make new graph curves to have better control. You then add mix curves as needed and select the mix curves in the control cards!

**Reply by godzinilla** [reply]
I understood your reply ;) My screenshot is jus the starting point for now with "corrected" fans speed with positive pressure setup.

If I set it as you described and set the 0 rpm fan option off, as the GPU cannot spins slower than 20% my minimal 13% from the CPU graph is not taken to consideration in the mixed (MAX) curve from the obvious reason:

![image](https://github.com/Rem0o/FanControl.Releases/assets/77450480/a3cf341c-f11c-4075-a373-afb18bccf589)

As my CPU fans are faster than INTAKE/EXHAUST fans (Light Wings High Speed vs regular; 2200 rpm vs 1500 rpm) the 20% for the CPU is faster than for the rest so I need to ramp the IN/EX fans to match the positive pressure assumptions. We all now what it means. It is not very loud, but it starting to be hearable.

**Reply by ionGL** [reply]
> I need an advice how to set the AIO fans to double act as a EXHAUST fan for the CPU and GPU temperature. 

This is where we started from. Where are your aio fans connected in your pictures? To the "Cpu Fan" control, right?

"GPU CUSTOM GRAPH" - here set the temp source to gpu core. Then open the graph and set the first point to 0% at 55C. Then the second point to 40% at 60C, or whatever % you feel it's needed.

When "GPU CUSTOM GRAPH" is mixed with "CPU" fan curves, and the mix is linked to the "CPU Fan" control, the Mix will decide what graph to follow. When the gpu gets hotter, it follows the "GPU CUSTOM GRAPH" curve. When the gpu temp drops bellow 55C, the "CPU" takes over and sets 13%.

Pay attention here, the "GPU CUSTOM GRAPH" has NOTHING to do with your radeon gpu fans! That "GPU CUSTOM GRAPH" sets the fan % only for the aio fans connected to the "CPU Fan" control.

**Reply by godzinilla** [reply]
Okay, I am getting you now. The **GPU CUSTOM GRAPH** is indeed my custom graph for controlling the GPU fans (there are some reasons to use it instead of the built in fan customization in the Adrenaline software). But you want me to create another graph based on the GPU temperature and use it only for controlling the AIO fans. It has a lot of sense now, sorry I have not supposed it before.

Anyway, I found a way to lower my GPU fans speed to the 13% which gave me about 700-800 RPMs instead of 1100 for the 20% - the difference is very noticeable. From this point I will try to create a graphs as you described or tests what I achieved now - the 13% for the GPU and the CPU is a good starting point for controlling the AIO based on these two temp sources.

Second thing to be done is to link the INTAKE and EXHAUST fan(s) like the AIO ones too ;)

**Reply by ionGL** [reply, marked as answer]
Yes, there it is! Each control works best if you have separate curves for them.

Redo your curves like this:

- Delete all auto and linear curves and make new **graph** curves and set them to RPM mode (3 dots top right on each curve)
- Make sure all fans are properly calibrated. Open the manual fan calibration window and set each fan to various % and make sure they stabilize. If you see RPMs that are not corresponding, adjust them as needed. Also add intermediary points, like 95%, 85%, 75% and so on.

![manaualfancalibration](https://github.com/Rem0o/FanControl.Releases/assets/80151979/d6f09b5b-3aa2-4eca-8bfe-1ff454935bdc)

- Use MIX curves when needed

Once you are done with the aio and case fans, do the same with the gpu fans. Give it a new RPM mode graph curve. If you feel confident, add a Trigger curve and mix it with the graph. The logic is that once the gpu temp got over a temp, the fans will be kept on with a low rpm to cool it back to a desired temp. 
Use my example below, where the gpu fans are controlled by a graph and a trigger. Once the temp gets over 53C, the fan starts spinning and follows the graph curve with the first point at 800 RPM and goes up as needed. As the load is done and the temp goes down, as it drops to 52C the fan doesn't turn off, it follows the "trigger" graph instructions and waits for 42C to turn it off. Nice, right?
![gpucurves](https://github.com/Rem0o/FanControl.Releases/assets/80151979/6ca6ad81-7fab-4410-8554-34fda7db35d0)

**Reply by godzinilla** [reply]
So I finally made it, but I need a one more advice ;)

Here it is:
![image](https://github.com/Rem0o/FanControl.Releases/assets/77450480/3df6b629-9645-42d4-8c8a-4d1dbd1847aa)

First of all I've switched to the RPM instead of percentage in my graphs. The **CPU FOR MIX** and **GPU FOR MIX** are for controlling all the fans. The **GPU CUSTOM RPM** is for the GPU fans only. I've checked and modified the calibration in 5% steps for all fans (except GPU). It works quite good now but mine last "issue" is how to efficiently manage the EXHAUST and AIO fans to keep them +/- the same RPMs keeping in mind that these are different fans (2200 vs 1500 RPM). I was trying to use the **Offset** function but it is not consistent because the different fans max speed. So I thinking to "trim" the AIO fans to the 1500 RPMs max in calibration - it is still good enough to cool all my peripherals. The last option is to switch the 1500 fans for the 2200 but it does not have sense while I won't be using them full throttle.

**Reply by ionGL** [reply]
Try adding a "Sync" curve, set it to RPM mode. Select CPU or EXHAUST for its control, then in the controls cards select the Sync curve for the one you didn't use in the Sync. Play around see if it works.

**Reply by godzinilla** [reply]
Adding the SYNC graph with the EXHAUST control, switching the CPU control to the SYNC graph and offsetting the INTAKE control a little does the trick, thank you! A little tweaking and I am home ;)

![image](https://github.com/Rem0o/FanControl.Releases/assets/77450480/b4bddb64-a165-4cd4-95e1-be255da774f2)

**Reply by ionGL** [reply]
Looking good. Remember how i told you about the option of having curves for each control card? You can go with that option. There will be a few more curves to manage, but once you set them, you have precise control for all fans, no need to use offset.

**Reply by godzinilla** [reply]
> Looking good. Remember how i told you about the option of having curves for each control card? You can go with that option. There will be a few more curves to manage, but once you set them, you have precise control for all fans, no need to use offset.

So you're suggesting to create a curve for the intake and exhaust fans and another one for the CPU too?

**Reply by ionGL** [reply]
Yes, that's exactly what i'm saying. Look on the right side. Excluding the Radeon card and the PUMP, you have **3 controls**.
If you want each control to respond to 2 temp sources, you add 2 graph + 1 mix curves for each control. 9 total + whatever you do for the radeon gpu. But there might be another way if we get a response from the developer.

@Rem0o, help us here with a clarification. Take a look at the latest picture posted by godzinilla. If all graphs are in RPM mode, should **one MIX** curve selected for multiple controls set a different fan % corresponding to the right calibrated RPM? The fans have different maps.
Or should he use MIX2, MIX3 assigned to each control, with the same graph curves?
